src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/TagHelperDescriptor.cs

```diff
@@ -204,7 +204,7 @@ public IEnumerable<RazorDiagnostic> GetAllDiagnostics()
         }
     }
 
-    public override string ToString()
+    public override string? ToString()
```

Comment:
For all these `?` additions to `ToString`, let's just remove them and ensure that we're never returning null from a `ToString` call.

Reply:
The problem is we call `base.ToString()` which is `object.ToString()` which returns `string?` - so we cannot just say we return `string` here, unless we use `return DisplayName ?? base.ToString()!;` or something but not sure we want that.

Reply:
Just suppress it. `object.ToString()` will never return `null`. I can give you the history lesson why it's annotated if you want, but the tldr is that it's expected to override and tighten the return nullability.